### PR TITLE
[TS] LPS-91113 No need to retrieve preferences if portlet id doesn't exist or is undeployed

### PR DESCRIPTION
**Pull request by marianoalvarosaiz**

### CODE REVIEW

**Comment by marianoalvarosaiz** [comment]
Hi @topolik,
 were you able to take a look to this pull?
Kind regards.

**Comment by topolik** [comment]
Hi, not yet, will process it today.

Thanks.

-- tom +

File: portal-impl/src/com/liferay/portal/service/impl/PortletPreferencesLocalServiceImpl.java [hunk 1]
```diff
@@ -509,12 +509,19 @@ public long getPortletPreferencesCount(
...
+					preferences = PortletConstants.DEFAULT_PREFERENCES;
+				}
+				else {
+					portletPreferences =
+						portletPreferencesPersistence.fetchByO_O_P_P(
+							ownerId, ownerType, plid, portletId);
+
+					if (portletPreferences != null) {
+						preferences = portletPreferences.getPreferences();
+					}
```
Comment (topolik) [inline portal-impl/src/com/liferay/portal/service/impl/PortletPreferencesLocalServiceImpl.java]: This is unnecessary. The `portlets` are taken from the layout's portlets, not from the request. Moreover, when portlet is not deployed it doesn't mean there are no preferences that cannot be processed for example by journal article template.

**Comment by topolik** [comment]
You also missed other places, can you please add a breakpoint here: https://github.com/liferay/liferay-portal/blob/bc39eb1e0fa400587119b98f23e81e0b96d9f768/portal-kernel/src/com/liferay/portal/kernel/dao/orm/QueryPos.java#L337

screenshot from 2019-02-28 09-37-13


And try for example with http://localhost:8080/web/guest/home?p_p_id=AAAA1&p_p_lifecycle=0&p_p_state=maximized

**Comment by topolik** [comment]
So I see 2 ways to fix this.

1, Fix the preferences API the way you fixed it + fix settings framework.
2, Fix callers of the preferences API and check for the undeployed portlet there.

Can you please resend? Thanks.

**Comment by marianoalvarosaiz** [comment]
see https://github.com/topolik/liferay-portal/pull/719

**Pull request closed without merging**